audio-worklet/src/noisegate-script-processor.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+class NoiseGate {
+  /**
+   * A Noise gate allows audio signals to pass only when the registered volume
+   * is above a specified threshold.
+   * @param  {BaseAudioContext} context the audio context
+   * @param  {Number} options.channels the number of input and output 
+   *                                   channels, a maximum of two
+   * @param  {Number} options.attack milliseconds for gate to fully close  
+   * @param  {Number} options.attack milliseconds for gate to fully open
```

Comment:
But I will change this to be consistent with web audio. By the same logic, is it better to use signal amplitude for the threshold rather than dB? Since that would involve the least amount of conversion back and forth between units when used in web audio?